backend/common/src/main/java/org/eclipse/sw360/datahandler/db/ConfigContainerRepository.java

```diff
@@ -37,17 +42,32 @@ public ConfigContainerRepository(DatabaseConnectorCloudant databaseConnector) {
     }
 
     public ConfigContainer getByConfigFor(ConfigFor configFor) {
-        PostViewOptions query = getConnector().getPostViewQueryBuilder(ConfigContainer.class, "byConfigFor")
-                .keys(Collections.singletonList(configFor.toString()))
-                .includeDocs(true).build();
-
-        List<ConfigContainer> configs = queryView(query);
-        if (configs.size() != 1) {
-            throw new IllegalStateException(
-                    "There are " + configs.size() + " configuration objects in the couch db for type " + configFor
-                            + " while there should be exactly one!");
-        } else {
+        try {
+            PostViewOptions query = getConnector().getPostViewQueryBuilder(ConfigContainer.class, "byConfigFor")
+                    .keys(Collections.singletonList(configFor.toString()))
+                    .includeDocs(true).build();
+
+            List<ConfigContainer> configs = queryView(query);
+
+            if (configs == null || configs.isEmpty()) {
+                log.warn("No ConfigContainer found for " + configFor + ". Using default config.");
+                return createDefaultFossologyConfig();
+            }
+
             return configs.get(0);
+        } catch (Exception e) {
+            log.error("Error fetching ConfigContainer for " + configFor, e);
+            return createDefaultFossologyConfig();
```

Comment:
Since #2815, the function is also called for `configFor == ConfigFor.SW360_CONFIGURATION`. Thus, the `createDefaultFossologyConfig()` should be called only for the `FOSSOLOGY_REST` config.